### DISCUSSION TITLE
Thread Name Truncated in Perfetto Trace

### DISCUSSION THREAD

**Discussion by KevinLiu707** [google/perfetto#960] [Q&A] [answered]
Hello Perfetto Team,

I am encountering an issue with thread names being truncated in Perfetto traces. Specifically, I have a thread named `com.example.myapplication.MainActivity`, but in the Perfetto trace output, it is displayed as `on.MainActivity`. This truncation makes it difficult to identify the threads correctly.

Could you provide guidance on how to display the full thread name in the Perfetto trace output?

- Show in Perfetto 

![Image](https://github.com/user-attachments/assets/b14df6d9-ae0b-4d17-966c-046d7df11278)

- Show in Android studio profiler
![Image](https://github.com/user-attachments/assets/dcd71c45-807d-4e0a-a06e-0059187db267)

- trace.json
[trace (1).json](https://github.com/user-attachments/files/17990978/trace.1.json)

Thank you for your assistance!

Best regards, Kevin.

**Comment by LalitMaganti** [comment, marked as answer]
The 16 character limitation is limitation of the underlying ftrace data source which the kernel provides for tracing. It is not a limitation of Perfetto itself

If full names are important to you, you can turn on the record_thread_names option in the process stats (https://perfetto.dev/docs/reference/trace-config-proto#ProcessStatsConfig) data source which will make us scrape /proc to get the full names. Note that this is somewhat expensive which is why we do not turn it on by default.

**Reply by KevinLiu707** [reply]
Hi, I tried following the instructions in the documentation by adding `record_thread_names: true `and removing` ftrace_events: "sched/sched_switch".` Despite multiple attempts, the thread names are still not displayed completely. Is my configuration file correct? 
<img width="1303" alt="image" src="https://github.com/user-attachments/assets/2d701e9c-f36e-46a4-b92e-a1867e281c41">
[config2.txt](https://github.com/user-attachments/files/18021334/config2.txt)

**Reply by LalitMaganti** [reply]
You should still keep sched_switch enabled. The config comment is a bit misleading, it's more referencing the fact that some thread names will show up.

**Reply by KevinLiu707** [reply]
I have already tried keeping sched_switch enabled, but still can't display the full thread names. If you have the correct command to capture the trace with full thread name , could you please send it to me?

**Reply by LalitMaganti** [reply]
Unless the thread is short lived, enabling `record_thread_names` is all that's needed.

**Reply by LalitMaganti** [reply]
If that's not working, I'm not sure what's wrong as I cannot repro this issue: we use this option occasionally and it works fine for us.

**Reply by KevinLiu707** [reply]
Thank you very much for your detailed response. I will try some other methods.

**Reply by LalitMaganti** [reply]
Actually I investigated this more now I got some time: I was misremembering when I said above. There is no option to collect full thread names. This is not a Perfetto problem, this is a limitation of the kernel itself. e.g. see https://stackoverflow.com/questions/5026531/thread-name-longer-than-15-chars

I'm not aware of any API in Linux which actually would give the full names, if you find one, let us know and we can add support for this.
